php/elFinderVolumeGoogleDrive.class.php

```
@@ -0,0 +1,1659 @@
+				$cache['width'] = $size[0];
+				$cache['height'] = $size[1];				
+				$ret = $size[0].'x'.$size[1];
+			}
+		}
+		is_file($work) && @unlink($work);
+		return $ret;
+	}
+
+	/******************** file/dir content *********************/
+
+	/**
+	 * Return files list in directory.
+	 *
+	 * @param  string  $path  dir path
+	 * @return array
+	 * @author Dmitry (dio) Levashov
+	 * @author Cem (DiscoFever)
+	 **/
+	protected function _scandir($path) {
+	
+		return isset($this->dirsCache[$path])
+			? $this->dirsCache[$path]
+			: $this->cacheDir($path);
+	}
+
+	/**
+	 * Open file and return file pointer
+	 *
+	 * @param  string  $path  file path
+	 * @param  bool    $write open file for writing
+	 * @return resource|false
+	 * @author Dmitry (dio) Levashov
+	 **/
+	protected function _fopen($path, $mode='rb') {
+	
+		if (($mode == 'rb' || $mode == 'r')) {
+			try {
+				//$files = new Google_Service_Drive_DriveFile();						
+				$itemId = basename($path);			
+							
+				$contents = $this->oauth->files->get($itemId, [					
+					//'fields' => 'id,webContentLink',
+					'alt' 	 => 'media'
+				]);
+				
+				$contents = $contents->getBody()->detach();
+				rewind($contents);	
+ 				//$fp = tmpfile();
+				//fputs($fp , $contents);
+				//while(trim(fgets($fp)) !== ''){};
+				//rewind($fp);				
+				return $contents;
+				
+			} catch (Exception $e) {
+				return false;
+			}
+		}
+		
+		if ($this->tmp) {
+			$contents = $this->_getContents($path);
+			
+			if ($contents === false) {
+				return false;
+			}
+			
+			if ($local = $this->getTempFile($path)) {
+				if (file_put_contents($local, $contents, LOCK_EX) !== false) {
+					return @fopen($local, $mode);
+				}
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Close opened file
+	 *
+	 * @param  resource  $fp  file pointer
+	 * @return bool
+	 * @author Dmitry (dio) Levashov
+	 **/
+	protected function _fclose($fp, $path='') {	
+		@fclose($fp);
+		if ($path) {
+			@unlink($this->getTempFile($path));
```

Do not use the `@` to hide the error.